### PR TITLE
CATTY-727 Implement minimal functionallity for SVG Plot creation

### PR DESCRIPTION
**Pull request by StofflR**
Implement minimal functionallity for SVG Plot creation
https://jira.catrob.at/browse/CATTY-727

Your checklist for this pull request

Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x]  Verify that the Jira ticket is in the status Ready for Development
- [x]  Include a summary of the changes plus the relevant context
- [x]  Choose the proper base branch (develop)
- [x]  Confirm that the changes follow the project’s coding guidelines
- [x]  Verify that the changes generate no compiler or linter warnings
- [x]  Perform a self-review of the changes
- [x]  Verify to commit no other files than the intentionally changed ones
- [x]  Include reasonable and readable tests verifying the added or changed behavior
- [x]  Confirm that new and existing unit tests pass locally
- [x]  Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x]  Stick to the project’s git workflow (rebase and squash your commits)
- [x]  Verify that your changes do not have any conflicts with the base branch
- [ ]  After the PR, verify that all CI checks have passed
- [ ]  Post a message in the #catty [Slack channel](https://catrobat.slack.com/) and ask for a code reviewer

Added basic functionality for basic svg creation for plotter
* start plot
* stop plot
* save plot to svg
Fixed PenUp/PenDown bug 
* separate arrays on penUp to retain the needed information

### CODE REVIEW

File: src/Catty/DataModel/Configuration/PenConfiguration.swift [hunk 1]
```diff
@@ -22,6 +22,8 @@
 
 struct PenConfiguration {
     var penDown = false
+    var cut = false
```
Comment (lucatp) [inline src/Catty/DataModel/Configuration/PenConfiguration.swift:25]: würde einen bool den Präfix is geben, also in dem fall isCut

File: src/Catty/Extension&Delegate&Protocol/Extensions/SpriteNode/CBSpriteNodePenExtension.swift [hunk 2]
```diff
@@ -23,25 +23,49 @@
 extension CBSpriteNode {
 
     @objc func drawPenLine() {
-
-        if penConfiguration.previousPositions.last != self.position && penConfiguration.penDown {
-            penConfiguration.previousPositions.append(self.position)
+        for i in 0..<self.penConfiguration.previousPositionLines.count
```
Comment (lucatp) [inline src/Catty/Extension&Delegate&Protocol/Extensions/SpriteNode/CBSpriteNodePenExtension.swift:26]: würde hier for element in penConfiguration.previousPositions machen anstatt einem for i loop

File: src/Catty/Extension&Delegate&Protocol/Extensions/SpriteNode/CBSpriteNodePenExtension.swift [hunk 3]
```diff
@@ -23,25 +23,49 @@
...
+        }
+        let positions = self.penConfiguration.previousPositions
+        drawLineFromConfiguration(with: positions , mode: false)
+        if positions.count > 1 {
+            self.penConfiguration.previousPositions.removeSubrange(0..<positions.count - 1)
+        }
+    }
+    
+    @objc func drawPlotLine() {
+        for i in 0..<self.penConfiguration.previousCutPositionLines.count
```
Comment (lucatp) [inline src/Catty/Extension&Delegate&Protocol/Extensions/SpriteNode/CBSpriteNodePenExtension.swift:42]: das gleiche wie oben

File: src/Catty/Extension&Delegate&Protocol/Extensions/SpriteNode/CBSpriteNodePenExtension.swift [hunk 4]
```diff
@@ -23,25 +23,49 @@
...
-            for (index, point) in penConfiguration.previousPositions.enumerated() where index > 0 {
-                guard let lineFrom = penConfiguration.previousPositions[index - 1] else {
+            for (index, point) in positions.enumerated() where index > startIndex && index > 0 {
+                guard let lineFrom = positions[index - 1] else {
                     fatalError("This should never happen")
                 }
                 let lineTo = point
-
-                self.addLine(from: lineFrom, to: lineTo, withColor: penConfiguration.color, withSize: penConfiguration.size)
+                if(dashed){
```
Comment (lucatp) [inline src/Catty/Extension&Delegate&Protocol/Extensions/SpriteNode/CBSpriteNodePenExtension.swift:62]: klammern brauchst beim if nicht

File: src/Catty/SavePlotSVGBrick+Instructions.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        if(!filename.hasSuffix(".svg")){
+            filename = filename + ".svg"
+        }
+        let file = self.getDocumentsDirectory().appendingPathComponent(filename)
+        let debugfile = self.getDocumentsDirectory().appendingPathComponent("filename.txt")
+        do {
+            try filecontent.write(to: file, atomically: true, encoding: String.Encoding.utf8)
+            try filecontent.write(to: debugfile, atomically: true, encoding: String.Encoding.utf8)
+        } catch {
+            // TODO: failed to write file
```
Comment (lucatp) [inline src/Catty/SavePlotSVGBrick+Instructions.swift:74]: ist kein error handling gewünscht? Wenn keines gewünscht ist kannst du auch try? schreiben dann brauchst kein do catch

**Review by lucatp — COMMENTED** [review]
Schaut so weit schon sehr gut aus, hab paar Sachen die man verbessern könnte dazu geschrieben.

File: src/Catty/SavePlotSVGBrick+Instructions.swift (hunk 5 above)
Comment (StofflR) [reply src/Catty/SavePlotSVGBrick+Instructions.swift:74]: Derweil is noch keines Gewünscht :)

File: src/Catty/XML/XMLHandler/Bricks/Plot/SavePlotSVGBrick+CBXMLHandler.swift [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+extension SavePlotSVGBrick: CBXMLNodeProtocol {
+    static func parse(from xmlElement: GDataXMLElement, with context: CBXMLParserContext) -> Self {
+        CBXMLParserHelper.validate(xmlElement, forFormulaListWithTotalNumberOfFormulas: 1)
+        let brick = self.init()
+        let formula = CBXMLParserHelper.formula(in: xmlElement, forCategoryName: "ASK_QUESTION", with: context)
```
Comment (wallisch) [inline src/Catty/XML/XMLHandler/Bricks/Plot/SavePlotSVGBrick+CBXMLHandler.swift:27]: `ASK_QUESTION` is probably not the right name, what is used on android?

**Review by wallisch — CHANGES_REQUESTED** [review]
I rebased on latest develop and corrected the swiftlint issues, can you please address the remaining comments?

**Review by wallisch — APPROVED** [review]

**Pull request merged**